### PR TITLE
Simplify overriding structure

### PR DESCRIPTION
**Pull request by pawelkisielewicz**
Resolves second (last) part of https://codepoets.atlassian.net/browse/GF-76

### CODE REVIEW

File: tests/golem/verificator/test_renderingverifier.py [hunk 1]
```diff
@@ -144,7 +135,8 @@ def test_simple_verification_frames_wrong_resolution(self):
         frame_rendering_verifier = FrameRenderingVerifier(
             self.verification_data
         )
-        frame_rendering_verifier.simple_verification()
+        result = frame_rendering_verifier.simple_verification()
         verifier_state = frame_rendering_verifier.verification_completed()[1]
 
+        self.assertFalse(result)
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py]: please adhere to the existing style of asserts:
```suggestion
        assert result is False
```

File: tests/golem/verificator/test_renderingverifier.py [hunk 2]
```diff
@@ -120,9 +110,10 @@ def test_simple_verification_frames_no_results(self):
...
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py]: please adhere to the existing style of asserts:
```suggestion
        assert result is False
```

File: tests/golem/verificator/test_renderingverifier.py [hunk 3]
```diff
@@ -109,9 +98,10 @@ def test_simple_verification_frames_less_tasks_than_frames(self):
...
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py]: please adhere to the existing style of asserts:
```suggestion
        assert result is False
```

File: tests/golem/verificator/test_renderingverifier.py [hunk 4]
```diff
@@ -76,30 +74,21 @@ def test_simple_verification_wrong_answer_when_result_is_not_an_image(self):
...
-            self.verification_data
-        )
-        frame_rendering_verifier.simple_verification()
-        verifier_state = frame_rendering_verifier.verification_completed()[1]
-
-        assert verifier_state == SubtaskVerificationState.VERIFIED
+    def test_simple_verification_returns_true_if_images_have_proper_sizes(self):
+        self.verification_data['results'] = self._create_images()
+        rendering_verifier = FrameRenderingVerifier(self.verification_data)
+        self.assertTrue(rendering_verifier.simple_verification())
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py]: please change assert style so it is coherent with the rest of the cod ein this module

File: tests/golem/verificator/test_renderingverifier.py [hunk 5]
```diff
@@ -53,7 +53,6 @@ def test_get_part_size(self):
             self.subtask_info) == (self.x, self.y)
 
     def test_simple_verification_wrong_answer_when_not_a_file(self):
-        # Result is not a file
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py:56]: :+1:

File: tests/golem/verificator/test_renderingverifier.py [hunk 6]
```diff
@@ -63,7 +62,6 @@ def test_simple_verification_wrong_answer_when_not_a_file(self):
         assert verifier_state == SubtaskVerificationState.WRONG_ANSWER
 
     def test_simple_verification_wrong_answer_when_result_is_not_an_image(self):
-        # Result is not an image
```
Comment (dybi) [inline tests/golem/verificator/test_renderingverifier.py:66]: :+1:

File: tests/golem/verificator/test_core_verifier.py [hunk 7]
```diff
@@ -34,31 +34,32 @@ def callback(result):
...
-        core_verifier.simple_verification()
-        assert core_verifier.state == expected_state
+        result = core_verifier.simple_verification()
+        self.assertIs(result, expected_result)
 
     def test_simple_verification(self):
 
         self.verification_data["results"] = []
-        self._check_state(SubtaskVerificationState.WRONG_ANSWER)
+        self._check_state(False)
```
Comment (dybi) [inline tests/golem/verificator/test_core_verifier.py]: please use it with explicit keyword argument (`self._check_state(expected_result=False)`) - the intent becomes more obvious

File: tests/golem/verificator/test_core_verifier.py [hunk 8]
```diff
@@ -34,31 +34,32 @@ def callback(result):
...
 
         assert sync_wait(deferred, 2) is True
 
-    def _check_state(self, expected_state: SubtaskVerificationState):
+    def _check_state(self, expected_result: bool):
         core_verifier = CoreVerifier(self.verification_data)
-        core_verifier.simple_verification()
-        assert core_verifier.state == expected_state
+        result = core_verifier.simple_verification()
+        self.assertIs(result, expected_result)
```
Comment (dybi) [inline tests/golem/verificator/test_core_verifier.py]: again, please use existing assertions style

File: golem/verificator/core_verifier.py [hunk 9]
```diff
@@ -16,38 +16,31 @@ class CoreVerifier:  # pylint: disable=too-many-instance-attributes
 
     def __init__(self, verification_data):
         self.verification_data = verification_data
-        self.subtask_info = self.verification_data['subtask_info']
-        self.resources = []
-        self.results = []
+        self.results = verification_data['results']
```
Comment (dybi) [inline golem/verificator/core_verifier.py:18]: it seems that after adding `self.results` and `self.subtask_info` fields, `self.verification_data` is not needed anymore...

File: golem/verificator/rendering_verifier.py [hunk 10]
```diff
@@ -35,13 +33,19 @@ def check_size(file_path, resolution_x, resolution_y):
...
 
-    def _verify_result(self, results):
-        subtask_info = results["subtask_info"]
-        results = results["results"]
-
-        resolution_x, resolution_y = self._get_part_size(subtask_info)
+    def simple_verification(self):
+        if not super().simple_verification():
+            return False
+        for _ in self.results:
```
Comment (dybi) [inline golem/verificator/rendering_verifier.py]: you don't need this `for` loop - tou iterate over results within `_are_image_sizes_correct()`

File: golem/verificator/rendering_verifier.py [hunk 11]
```diff
@@ -11,10 +11,8 @@ class RenderingVerifier(CoreVerifier):
 
     def __init__(self, verification_data):
         super().__init__(verification_data)
-        self.verification_data = verification_data
-        self.resources = verification_data['resources']
-        self.results = verification_data['results']
         self.state = SubtaskVerificationState.WAITING
+        self.resources = verification_data['resources']
```
Comment (dybi) [inline golem/verificator/rendering_verifier.py]: it seems that `self.resources` are not used anywhere... neither in this class nor in `CoreVerifier`

File: golem/verificator/rendering_verifier.py [hunk 12]
```diff
@@ -35,13 +33,19 @@ def check_size(file_path, resolution_x, resolution_y):
...
+                self.message = 'No proper task result found'
+                self.state = SubtaskVerificationState.WRONG_ANSWER
+                return False
+        return True
 
-        for image in results:
+    def _are_image_sizes_correct(self):
+        resolution_x, resolution_y = self._get_part_size(self.subtask_info)
+        for image in self.results:
             if not self.check_size(image, resolution_x, resolution_y):
```
Comment (dybi) [inline golem/verificator/rendering_verifier.py:47]: in `check_size()` only x dimension is checked... what about y?

File: tests/golem/verificator/test_core_verifier.py [hunk 13]
```diff
@@ -34,31 +34,32 @@ def callback(result):
             assert state == SubtaskVerificationState.VERIFIED
             deferred.callback(True)
 
+        self.core_verifier.simple_verification()
```
Comment (dybi) [inline tests/golem/verificator/test_core_verifier.py]: why is it called (and result is ignored)?

**Review by dybi — CHANGES_REQUESTED** [review]
There is some doubts to clarify and some small improvements to be made.

File: tests/golem/verificator/test_renderingverifier.py (hunk 1 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_renderingverifier.py]: done

File: tests/golem/verificator/test_renderingverifier.py (hunk 2 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_renderingverifier.py]: done

File: tests/golem/verificator/test_renderingverifier.py (hunk 3 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_renderingverifier.py]: done

File: tests/golem/verificator/test_renderingverifier.py (hunk 4 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_renderingverifier.py]: done

File: tests/golem/verificator/test_renderingverifier.py (hunk 5 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_renderingverifier.py:56]: :)

File: tests/golem/verificator/test_core_verifier.py (hunk 7 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_core_verifier.py]: done

File: tests/golem/verificator/test_core_verifier.py (hunk 8 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_core_verifier.py]: done

File: golem/verificator/core_verifier.py (hunk 9 above)
Comment (pawelkisielewicz) [reply golem/verificator/core_verifier.py:18]: It was used, but it was easy to change this to other fields and don't use any more. Good catch :+1:

File: golem/verificator/rendering_verifier.py (hunk 10 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py]: good catch, done

File: golem/verificator/rendering_verifier.py (hunk 11 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py]: done

File: golem/verificator/rendering_verifier.py (hunk 12 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py:47]: good question, didn't payd attention to it. Fixed

File: tests/golem/verificator/test_core_verifier.py (hunk 13 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_core_verifier.py]: it was used before, but after final changes is not required any more. Deleted

File: golem/verificator/rendering_verifier.py (hunk 12 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py:47]: After this change there is failing test `TestBlenderFrameTask.test_computation_failed_or_finished`. I investigated this and problem is that method `_get_part_size` in blender_verifier returns resolution y = 0. @dybi you recently wrote tests for this method. Could you run this tests and help me to investigate, where is the problem?

File: golem/verificator/blender_verifier.py [hunk 14]
```diff
@@ -77,10 +77,10 @@ def failure(exception):
         self.finished.addCallback(success)
         self.finished.addErrback(failure)
 
-        subtask_info = self.verification_data['subtask_info']
-        work_dir = os.path.dirname(self.verification_data['results'][0])
+        # subtask_info = self.verification_data['subtask_info']
```
Comment (dybi) [inline golem/verificator/blender_verifier.py]: please remove comment

File: golem/verificator/core_verifier.py [hunk 15]
```diff
@@ -15,39 +15,31 @@ class CoreVerifier:  # pylint: disable=too-many-instance-attributes
...
-        for result in results:
-            if not os.path.isfile(result) or not\
-                    self._verify_result(self.verification_data):
+        for result in self.results:
+            if not os.path.isfile(result):
                 self.message = 'No proper task result found'
                 self.state = SubtaskVerificationState.WRONG_ANSWER
                 return False
-
-        self.state = SubtaskVerificationState.VERIFIED
```
Comment (dybi) [inline golem/verificator/core_verifier.py:50]: Was it deleted accidentally or else what would be the reason?

File: golem/verificator/rendering_verifier.py [hunk 16]
```diff
@@ -53,16 +55,12 @@ def simple_verification(self):
...
+        frames = self.subtask_info['all_frames']
         if use_frames and total_tasks <= len(frames):
-            frames_list = subtask_info['frames']
-            if len(results) < len(frames_list):
+            frames_list = self.subtask_info['frames']
+            if len(self.results) < len(frames_list):
                 self.state = SubtaskVerificationState.WRONG_ANSWER
                 return False
-
-        self.state = SubtaskVerificationState.VERIFIED
```
Comment (dybi) [inline golem/verificator/rendering_verifier.py:67]: same here - shouldn't it stay?

File: tests/golem/verificator/test_core_verifier.py [hunk 17]
```diff
@@ -39,26 +39,26 @@ def callback(result):
 
         assert sync_wait(deferred, 2) is True
 
-    def _check_state(self, expected_state: SubtaskVerificationState):
+    def _check_state(self, expected_result: bool):
         core_verifier = CoreVerifier(self.verification_data)
-        core_verifier.simple_verification()
-        assert core_verifier.state == expected_state
+        result = core_verifier.simple_verification()
+        assert result == expected_result
```
Comment (dybi) [inline tests/golem/verificator/test_core_verifier.py]:
```suggestion
        assert result is expected_result
```

File: golem/verificator/rendering_verifier.py (hunk 12 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py:47]: as discussed, done in separate commit

**Review by dybi — APPROVED** [review]
bapprove

File: golem/verificator/blender_verifier.py (hunk 14 above)
Comment (pawelkisielewicz) [reply golem/verificator/blender_verifier.py]: done

File: golem/verificator/core_verifier.py (hunk 15 above)
Comment (pawelkisielewicz) [reply golem/verificator/core_verifier.py:50]: state `VERIFIED` is set in `start_verification` after positive result of `simple_verification`. It should be deleted from here

File: golem/verificator/rendering_verifier.py (hunk 16 above)
Comment (pawelkisielewicz) [reply golem/verificator/rendering_verifier.py:67]: no, explanation is above

File: tests/golem/verificator/test_core_verifier.py (hunk 17 above)
Comment (pawelkisielewicz) [reply tests/golem/verificator/test_core_verifier.py]: right, done

**Pull request merged**